include/seastar/testing/perf_tests.hh

```diff
@@ -167,23 +167,26 @@ class time_measurement {
     perf_stats _start_stats;
     perf_stats _total_stats;
 
+    linux_perf_event* _instructions_retired_counter;
+
 public:
     [[gnu::always_inline]] [[gnu::hot]]
     void start_run(linux_perf_event* instructions_retired_counter = nullptr) {
+        _instructions_retired_counter = instructions_retired_counter;
         _total_time = { };
         auto t = clock_type::now();
         _run_start_time = t;
         _start_time = t;
-        _start_stats = perf_stats::snapshot(instructions_retired_counter);
+        _start_stats = perf_stats::snapshot(_instructions_retired_counter);
     }
 
     [[gnu::always_inline]] [[gnu::hot]]
-    performance_test::run_result stop_run(linux_perf_event* instructions_retired_counter = nullptr) {
+    performance_test::run_result stop_run() {
         auto t = clock_type::now();
         performance_test::run_result ret;
         if (_start_time == _run_start_time) {
             ret.duration = t - _start_time;
-            auto stats = perf_stats::snapshot(instructions_retired_counter);
+            auto stats = perf_stats::snapshot(_instructions_retired_counter);
```

Comment:
let's reset `_instructions_retired_counter` to nullptr after it's used in stop_run since the caller is about to disable it.

Otherwise, I'm not thrilled about introducing this state in `time_measurement`, but I too see no better way of achieving this for `{start,stop}_measuring_time`.

Reply:
Done, I've nulled out the pointer in `stop_run()` and also set it to null in the constructor, to preserve the invariant that the pointer is only non-null during the run (if passed a non-null pointer in start) and null otherwise.